### ISSUE TITLE
Fix `selector-pseudo-element-no-unknown` false positives for `::highlight`

### ISSUE THREAD

**Issue by jathak** [stylelint/stylelint#6366]
### What steps are needed to reproduce the bug?

Code like

```
::highlight(a) {
  background-color: green;
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-element-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Online Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcD4AsCWBzZAbDyAuAFAIYCUABMADoB2ZZARkWANboBOEArtVALSRYIbeGXYwY1ANw0AviAA0IAM4BPaniIAPBOCVKF4CNQBmGHVVplKINpywwl1kRbp1rSmPbB4hvAA4enFAQvJ4wALYSeLzUIdzMsQDu1E5keLYwNHQysiAyQA)

### Which version of Stylelint are you using?

`14.12.1`

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problem was reported:

```
1:1-12    Unexpected unknown pseudo-element selector "::highlight" (selector-pseudo-element-no-unknown)
```

This pseudo-element is part of the [CSS Custom Highlight API](https://www.w3.org/TR/css-highlight-api-1/) and was [shipped in Chrome 105](https://chromiumdash.appspot.com/commit/68b3e7216572da1512f10f0d16351ef0b0f3532b).

### Proposal to fix the bug

PR incoming

**Comment by ybiquitous** [comment]
@jathak Thanks for providing the precise reproduction. I've copied and pasted an "Online Demo" link in this issue description. Also, thank you for creating a quick-fix PR. I'll soon review it.
